### PR TITLE
Added case and article ids to AttachResult custom events

### PR DESCRIPTION
**Pull request by mikegron**
Small addition to the AttachResult component to specify the caseId and the field where the articleId can be obtained. For ServiceNow, the articleId is pretty much always the permanentID and the caseID is static on the page. Tested using the local test page.

- Added case and article ids to AttachResult custom events
- Added click event when attaching in AttachResult

SNOW-364

### CODE REVIEW

File: src/components/AttachResult/AttachResult.ts [hunk 1]
```diff
@@ -50,6 +54,10 @@ export class AttachResult extends Component {
         detachCaption: ComponentOptions.buildStringOption({
             defaultValue: l(`${AttachResult.ID}_Detach`)
         }),
+        articleIdField: ComponentOptions.buildStringOption({
+            defaultValue: 'permanentid'
```
Comment (louis-bompart) [inline src/components/AttachResult/AttachResult.ts:58]: Why should it take this default value?

File: src/components/AttachResult/AttachResult.ts [hunk 2]
```diff
@@ -194,8 +203,11 @@ export class AttachResult extends Component {
         let customData: IAnalyticsCaseAttachMeta = {
             resultUriHash: this.queryResult.raw.urihash,
             author: this.queryResult.raw.author,
-            articleID: null,
-            caseID: null
+            articleID:
+                this.options.articleIdField && this.queryResult.raw[this.options.articleIdField]
+                    ? this.queryResult.raw[this.options.articleIdField]
+                    : null,
```
Comment (louis-bompart) [inline src/components/AttachResult/AttachResult.ts:209]: Can be simplified like so:
`articleID: this.queryResult.raw[this.options.articleIdField] ?? null`
I think.

File: src/components/AttachResult/AttachResult.ts [hunk 3]
```diff
@@ -194,8 +203,11 @@ export class AttachResult extends Component {
...
+            caseID: this.options.caseId
```
Comment (lbergeron) [inline src/components/AttachResult/AttachResult.ts:210]: Is there are reason why you use a different casing here compared to the option name? I would use the same casing everywhere.

`caseID` -> `caseId`
`articleID` -> `articleId`

Comment (mikegron) [reply src/components/AttachResult/AttachResult.ts:210]: je travaille sur les tests! :D
le casing cétait pour que sa fit avec les alias dans les dom attributes, sinon sa ferais `data-case-i-d`

File: src/components/AttachResult/AttachResult.ts (hunk 1 above)
Comment (mikegron) [reply src/components/AttachResult/AttachResult.ts:58]: that's the default ML uses to compute things so it's the most relevant field normally (ml-wise)

File: src/components/AttachResult/AttachResult.ts (hunk 3 above)
Comment (lbergeron) [reply src/components/AttachResult/AttachResult.ts:210]: Pour le casing, j'irais comme ceci. C'est bien ce que tu avais en tête?

    caseId: this.options.caseId

Comment (lbergeron) [reply src/components/AttachResult/AttachResult.ts:210]: Je viens de voir que `articleID` et `caseID` étaient déjà définies. Laisses le casing tel qu'il est alors. J'étais sur l'impression que c'était de nouveaux attributs.

**Pull request merged**